client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java

```diff
@@ -713,8 +713,10 @@ public void operationFail(Throwable throwable) {
         } catch (Exception ex) {
             long cost = System.currentTimeMillis() - beginStartTime;
             producer.updateFaultItem(brokerName, cost, true, false);
-            onExceptionImpl(brokerName, msg, timeoutMillis - cost, request, sendCallback, topicPublishInfo, instance,
-                retryTimesWhenSendFailed, times, ex, context, true, producer);
+            if (!(ex instanceof RemotingTooMuchRequestException)) {
+                onExceptionImpl(brokerName, msg, timeoutMillis - cost, request, sendCallback, topicPublishInfo, instance,
+                        retryTimesWhenSendFailed, times, ex, context, true, producer);
+            }
```

Comment:
Even if not retried, the onExceptionImpl method should still be executed, with the retry parameter set to false.

Reply:
Got，I will make some change for this.